force-app/main/default/classes/FileListViewController.cls

```diff
@@ -0,0 +1,60 @@
+public with sharing class FileListViewController {
+	@AuraEnabled
+    public static List<File> getContentDocuments(String pageSize, String pageNumber, String libraryFilter) {     
+        private Integer pSize = Integer.valueOf(pageSize);
+        private Integer pNumber = Integer.valueOf(pageNumber)-1;
+        private String likeLibraryFilter = '%' + libraryFilter + '%';
+        private List<ContentWorkspaceDoc> workspaces = new List<ContentWorkspaceDoc>();
+        
+        if(string.isEmpty(libraryFilter)) {
+            workspaces = [SELECT ContentDocument.Id, ContentWorkspace.Id, ContentDocument.Owner.Id, ContentDocument.Title, ContentWorkspace.Name, ContentDocument.Owner.Name, ContentDocument.LastModifiedDate 
+							FROM ContentWorkspaceDoc LIMIT :Integer.valueOf(pSize) OFFSET :(pSize*pSize)];
+        } else {
+            workspaces =  [SELECT ContentDocument.Id, ContentWorkspace.Id, ContentDocument.Owner.Id, ContentDocument.Title, ContentWorkspace.Name, ContentDocument.Owner.Name, ContentDocument.LastModifiedDate 
+							FROM ContentWorkspaceDoc WHERE ContentWorkspace.Name LIKE :likeLibraryFilter LIMIT :Integer.valueOf(psize) OFFSET :(psize*pnumber)];
+        }
+        List<File> files = new List<File>();
+        
+        for(ContentWorkspaceDoc w : workspaces) {
+            File file = new File(
+                w.ContentDocument.Id,
+                w.ContentWorkspace.Id,
+                w.ContentDocument.Owner.Id,
+            	w.ContentDocument.Title,
+                w.ContentWorkspace.Name,
+                w.ContentDocument.Owner.Name,
+                w.ContentDocument.LastModifiedDate
+            );
+            files.add(file);
+         }
+        return files;
+    }
+	
+    @AuraEnabled
+    public static List<ContentWorkspace> getSelectOptions() {
+        return [SELECT Name FROM ContentWorkspace];
```

Comment:
We should use either a filter condition or a LIMIT when making a SELECT. We shouldn't have queries With No Where Or Limit Clause.